src/main.rs

```diff
@@ -12,23 +16,36 @@ struct Opts {
     #[clap(short, long, default_value = "function.wasm")]
     function: PathBuf,
 
-    /// Path to json file containing Function input
-    input: PathBuf,
+    /// Path to json file containing Function input or piped input
+    input: Option<PathBuf>,
 
     /// Log the run result as a JSON object
-    #[clap(short, long, parse(from_flag))]
+    #[clap(short, long)]
     json: bool,
 }
 
 fn main() -> Result<()> {
     let opts: Opts = Opts::parse();
 
-    let function_run_result = run(opts.function, opts.input)?;
+    let mut input: Box<dyn Read + Sync + Send + 'static> = if let Some(ref input) = opts.input {
+        Box::new(BufReader::new(File::open(input).map_err(|e| {
+            anyhow!("Couldn't load input {:?}: {}", input, e)
+        })?))
+    } else {
+        Box::new(BufReader::new(stdin()))
+    };
+
+    let mut buffer = Vec::new();
+    input.read_to_end(&mut buffer)?;
+    let _ = serde_json::from_slice::<serde_json::Value>(&buffer)
+        .map_err(|e| anyhow!("Invalid input JSON: {}", e))?;
+
+    let function_run_result = run(opts.function, buffer);
```

Comment:
I think this will show the error case nicely and mean that we don't have to unwrap below?
```suggestion
    let function_run_result = run(opts.function, buffer)?;
```